cpp/modmesh/linalg/factorization.hpp

```diff
@@ -56,45 +56,76 @@ class Llt
 
 private:
 
-    // Forward substitution: solve L y = b
-    static array_type forward_substitution(array_type const & L, array_type const & b);
-    // Backward substitution: solve L^T x = y (or L^H x = y for complex)
-    static array_type backward_substitution(array_type const & L, array_type const & y);
+    // Forward substitution: solve L Y = B
+    static array_type forward_substitution(array_type const & L, array_type const & B);
+    // Backward substitution: solve L^T X = Y
+    static array_type backward_substitution(array_type const & L, array_type const & Y);
 
 }; /* end class Llt */
 
 template <typename T>
-auto Llt<T>::forward_substitution(array_type const & L, array_type const & b) -> array_type
+auto Llt<T>::forward_substitution(array_type const & L, array_type const & B) -> array_type
 {
     const size_t n = L.shape(0);
-    array_type y(n);
-    for (size_t i = 0; i < n; ++i)
+    const size_t m = B.shape(1);
```

Comment:
Incorrect logic to read `B.shape(1)` before testing `B.ndim() != 2`.

You also need to add a unit test for the error message for the input of wrong dimension.